src/app/pages/common/entity/entity-form/entity-form.component.ts

```diff
@@ -557,12 +557,15 @@ export class EntityFormComponent implements OnInit, OnDestroy, OnChanges, AfterV
     return this.fb.control({disabled, value}, validation);
   }
 
-  setDisabled(name: string, disable: boolean, hide: boolean = false, status?:string) {
+  setDisabled(name: string, disable: boolean, hide?: boolean, status?:string) {
     // if field will be hide, disabled it too
     if (hide) {
       disable = hide;
+    } else {
+      hide = false;
```

Comment:
Sorry, don't understand the reason for this.  hide is already false or the if would not get here.  Unless this is to set it to a particular known value (false rather than zero, say).

Reply:
'hide' was being set to 'false' as a parameter to setDisabled() which meant that the condition 'if(hide)' would never evaluate to true. So I made it an optional parameter, which means that if it isn't set in the component, then it is 'undefined' when it gets here, so I explicitly set it to 'false'.

We could eliminate all of these lines if we made sure every place calling setDisabled, passed in all the arguments.